Editor/AGS.Editor/BuildTargets/BuildTargetDataFile.cs

```diff
@@ -59,7 +64,78 @@ private string[] ConstructFileListForDataFile()
                 }
             }
             Utilities.AddAllMatchingFiles(files, "*.ogv");
-            return files.ToArray();
+
+            // Regular files are registered under their filenames (w/o dir)
```

Comment:
`assetList` gets iterated over twice, once in foreach and second in `ToArray()`. It's kind of wasteful so maybe try to avoid it with lazy executing code, there's 2 ways to do that, Linq or yield keyword.

Additionally it might be a good idea to change `ConstructCustomFileListForDataFile(...)` to return empty collection on failure instead of `null` because then you can concat it

Linq
```csharp
return files
    .Select(f => new Tuple<string, string>(Path.GetFileName(f), f));
    .Concat(ConstructCustomFileListForDataFile(errors))
    .ToArray();
```

yield, in order for this to work the return value needs to be changed to `IEnumerable<Tuple<string, string>>`
```csharp
foreach (var f in files)
    yield return new Tuple<string, string>(Path.GetFileName(f), f);

foreach (var f in ConstructCustomFileListForDataFile(errors))
    yield return f;
```